examples/tilted_bottom_boundary_layer.jl

```diff
@@ -0,0 +1,189 @@
+# the spacing near the bottom is relatively uniform with height, which is a desired property from
+# a numerical perspective.
+#
+#
+# ## Setting up a buoyancy model in a tilted domain
+#
+# We set-up our domain in a way that the coordinates align with the bottom. That means that, from
+# the coordinate's point of view, gravity needs to be tilted by `θ_rad` radians, which we can do by
+# passing the `vertical_units_vector` parameter to `Buoyancy()`:
+
+ĝ = [sin(params.θ_rad), 0, cos(params.θ_rad)]
+buoyancy = Buoyancy(model=BuoyancyTracer(), vertical_unit_vector=ĝ)
+
+# Since we are simulating a period domain that is tilted, we need to set-up the background
+# stratification as a `BackgroundField` and solve for the buoyancy perturbation. This avoids mean
+# horizontal buoyant gradients, which cannot exist in a horizontally-periodic domain:
+
+b∞(x, y, z, t, p) = p.N²∞ * (x * sin(p.θ_rad) + z * cos(p.θ_rad))
+B_field = BackgroundField(b∞, parameters=(; params.N²∞, params.θ_rad))
+
+
+# ## Bottom drag
+#
+# We also set-up a bottom drag that follows Monin-Obukhov theory, which can be implemented following https://doi.org/10.1029/2005WR004685.
+# Note that we need to include`V∞` in the velocity, since we will model it as a background field.
+#
+
+const κ = 0.4 # von Karman constant
+z₁ = CUDA.@allowscalar znodes(Center, grid)[1] # Closest grid center to the bottom
+cᴰ = (κ / log(z₁/params.z₀))^2 # Drag coefficient
+
+@inline drag_u(x, y, t, u, v, p) = - p.cᴰ * √(u^2 + (v + p.V∞)^2) * u
+@inline drag_v(x, y, t, u, v, p) = - p.cᴰ * √(u^2 + (v + p.V∞)^2) * (v + p.V∞)
+
+drag_bc_u = FluxBoundaryCondition(drag_u, field_dependencies=(:u, :v), parameters=(cᴰ=cᴰ, V∞=params.V∞))
+drag_bc_v = FluxBoundaryCondition(drag_v, field_dependencies=(:u, :v), parameters=(cᴰ=cᴰ, V∞=params.V∞))
+
+u_bcs = FieldBoundaryConditions(bottom = drag_bc_u)
+v_bcs = FieldBoundaryConditions(bottom = drag_bc_v)
+
+V_bg(x, y, z, t, p) = p.V∞
+V_field = BackgroundField(V_bg, parameters=(; V∞=params.V∞))
+#-----
+
+
+
+# ## Create the `NonhydrostaticModel`
+#
+# We create model with a `UpwindBiasedFifthOrder` advection scheme and a `RungeKutta3` timestepper. For simplicity,
+# we use a constant-diffusivity turbulence closure:
+
+model = NonhydrostaticModel(grid = grid, timestepper = :RungeKutta3,
+                            architecture = arch,
+                            advection = UpwindBiasedFifthOrder(),
+                            buoyancy = buoyancy,
+                            coriolis = ConstantCartesianCoriolis(f=params.f₀, rotation_axis=ĝ),
+                            tracers = :b,
+                            closure = IsotropicDiffusivity(ν=params.ν, κ=params.ν),
+                            boundary_conditions = (u=u_bcs, v=v_bcs,),
+                            background_fields = (b=B_field, v=V_field,),
+                           )
+
+
+# ## Create a simulation
+#
+#
+
+using Oceananigans.Grids: min_Δz
+if ndims==3
+    cfl=0.85
+else
+    cfl=0.5
+end
```

Comment:
For the sake of the example should we just use `cfl=0.5`?

Reply:
Oh, yes, totally. I'm still cleaning the code. That's why it's still a draft